### ISSUE TITLE
Rollup doesn't detect the redefinition of `clearTimeout`

### ISSUE THREAD

**Issue by mindrones** [rollup/rollup#1086]
Hi, I'm having some troubles building the linked example (a simple github search) with Rollup, while the Browserify build works as expected.

To reproduce:
- download [this gist](https://gist.github.com/mindrones/4a68fb4484600ed50cd6934c5f1c5c16);
- run `npm install`;
- run `npm run bundles`: this builds 2 bundles, one with Rollup, one with Browserify;
- run `python -m SimpleHTTPServer` (should serve from port 8000);
- open [http://localhost:8000/index_browserify.html](http://localhost:8000/index_browserify.html) and [http://localhost:8000/index_rollup.html](http://localhost:8000/index_rollup.html);
- in both pages type 'rollup' and press 'Go':
  - `index_browserify.html` executes the GET correctly and shows a list of links (correct);
  - `index_rollup.html` executes the GET correctly but then gives an error `Uncaught TypeError: Cannot set property '_timeout' of undefined`, probably caused by the fact that `node_modules/superagent/lib/request-base.js` exports `clearTimeout`:
    
    ``` js
    exports.clearTimeout = function _clearTimeout(){
      this._timeout = 0;
      clearTimeout(this._timer);
      return this;
    };
    ```
    
    but I'm not sure exactly, I'd need to investigate more. I've tried using named exports with no success.

Any hints about how to fix this? Thanks!

**Comment by thisconnect** [comment]
random quick guess can you try `commonjs({ include: 'node_modules/**' })`?

**Comment by mindrones** [comment]
Thanks, got the same error :/

**Comment by btd** [comment]
It looks very similar to https://github.com/rollup/rollup/issues/1007

**Comment by mindrones** [comment]
You're right, it's the same issue and indeed replacing `clearTimeout` with `window.clearTimeout` (as suggested by @ahelmberger in that issue),

```js
replace({
  include: '**/superagent/lib/request-base.js',
  values: { 'clearTimeout': 'window.clearTimeout' }
}),
```
works but I'd consider patching a library like this a workaround as you never know the side effects you are introducing; also, people might not realize the problem at hand and get blocked in similar situations :/

**Comment by mindrones** [comment]
Confirming that using `moduleContext` doesn't work, at least using:

```
moduleContext: {
  './node_modules/superagent/lib/request-base.js': 'window'
},
```

**Comment by mindrones** [comment]
Also, this `superagent` patch by @ahelmberger [fixes the problem but it's not released yet](https://github.com/visionmedia/superagent/pull/1088#issuecomment-257077749), so in case you try this gist and everything goes fine, it might be that `superagent` has been released meanwhile.

**Comment by ahelmberger** [comment]
It's in [Alpha](https://github.com/visionmedia/superagent/releases/tag/3.0.0-alpha.1) now!

**Comment by Victorystick** [comment]
@mindrones Mind if I close this as `superagent` is being fixed?

**Comment by mindrones** [comment]
@Victorystick but what if some other library does the same as `superagent`?
Do you think it's feasible for `rollup` to detect such cases?

**Comment by btd** [comment]
Probably it should (maybe just check global[name] for collisions) rename global collisions to something else (adding prefix or something).

**Comment by ahelmberger** [comment]
Strictly speaking, `superagent` was never broken ;-) It's certainly an edge case, but IMHO rollup should be able to deal with it, if possible.

**Comment by Rich-Harris** [comment]
Fixed with rollup-plugin-commonjs 6.0.1 – thanks
